src/box/tuple_format.c

```diff
@@ -1003,21 +1026,33 @@ tuple_field_map_create_plain(struct tuple_format *format, const char *tuple,
 	struct region *region = &fiber()->gc;
 	const char *pos = tuple;
 	uint32_t defined_field_count = mp_decode_array(&pos);
+	uint32_t format_field_count = tuple_format_field_count(format);
+	uint32_t common_field_count = MIN(defined_field_count,
+					  format_field_count);
+
 	if (validate && format->exact_field_count > 0 &&
 	    format->exact_field_count != defined_field_count) {
-		diag_set(ClientError, ER_EXACT_FIELD_COUNT,
-			 (unsigned) defined_field_count,
-			 (unsigned) format->exact_field_count);
-		return -1;
+		if (defined_field_count > format->exact_field_count)
+			goto error_exact_field_count;
+		if (!format->has_field_defaults)
+			goto error_exact_field_count;
+		/*
+		 * Do not complain about missed fields that have default values.
```

Comment:
`tuple_field_map_create_plain()` was called from here:
https://github.com/tarantool/tarantool/blob/ae02f0cd37af634db9ed2b3aed3e0604b34120ec/src/box/space.c#L573-L574
But that change is no longer needed, because now `space_apply_defaults()` is called before `space_before_replace()`.